### PR TITLE
Moves server packages into zipkin2 to simplify logging

### PR DESCRIPTION
**Pull request by codefromthecrypt**
Currently, we have to specify dual roots for logging categories due to
historical use of mixed types. This moves everything into the same
package which is not only a win for consistency, but also allows simpler
logging configuration.

Note `zipkin.server.ZipkinServer` and `zipkin.server.EnableZipkinServer`
are retained to not break people launching via java directly.

See #2047

### CODE REVIEW

File: zipkin-autoconfigure/collector-kafka/src/main/java/zipkin2/autoconfigure/collector/kafka/ZipkinKafkaCollectorAutoConfiguration.java [hunk 1]
```diff
@@ -11,7 +11,7 @@
  * or implied. See the License for the specific language governing permissions and limitations under
  * the License.
  */
-package zipkin.autoconfigure.collector.kafka10;
+package zipkin2.autoconfigure.collector.kafka;
```
Comment (codefromthecrypt) [inline zipkin-autoconfigure/collector-kafka/src/main/java/zipkin2/autoconfigure/collector/kafka/ZipkinKafkaCollectorAutoConfiguration.java:14]: note: all the autoconfigure types were package protected, meaning no-one could make a hard code dep on them without craziness

File: zipkin-autoconfigure/storage-elasticsearch/src/main/resources/META-INF/spring.factories [hunk 2]
```diff
@@ -1,4 +1,4 @@
 org.springframework.boot.autoconfigure.EnableAutoConfiguration=\
-zipkin.autoconfigure.storage.elasticsearch.http.ZipkinElasticsearchOkHttpAutoConfiguration,\
-zipkin.autoconfigure.storage.elasticsearch.http.ZipkinElasticsearchStorageAutoConfiguration,\
-zipkin.autoconfigure.storage.elasticsearch.http.TracingZipkinElasticsearchStorageAutoConfiguration
+zipkin2.autoconfigure.storage.elasticsearch.ZipkinElasticsearchOkHttpAutoConfiguration,\
+zipkin2.autoconfigure.storage.elasticsearch.ZipkinElasticsearchStorageAutoConfiguration,\
```
Comment (codefromthecrypt) [inline zipkin-autoconfigure/storage-elasticsearch/src/main/resources/META-INF/spring.factories:3]: this one is public, but I have no problem creating a change in zipkin-aws to accomodate

File: zipkin-server/README.md [hunk 3]
```diff
@@ -51,7 +51,7 @@ By default, zipkin writes log messages to the console at INFO level and above. Y
 For example, if you want to enable debug logging for all zipkin categories, you can start the server like so:
 
 ```bash
-$ java -jar zipkin.jar --logging.level.zipkin=DEBUG --logging.level.zipkin2=DEBUG
+$ java -jar zipkin.jar --logging.level.zipkin2=DEBUG
```
Comment (codefromthecrypt) [inline zipkin-server/README.md:54]: here's the actual important side effect

**Review by devinsba — COMMENTED** [review]
LGTM 👍

**Pull request merged**